contracts/core/Allo.sol

```diff
@@ -1,79 +1,368 @@
+    }
+
+    /// @notice Updates the treasury address
+    /// @param _treasury The new treasury address
+    /// @dev Only callable by the owner
+    function updateTreasury(address payable _treasury) external onlyOwner {
+        treasury = _treasury;
+        emit TreasuryUpdated(treasury);
+    }
+
+    /// @notice Updates the fee percentage
+    /// @param _feePercentage The new fee
+    /// @dev Only callable by the owner
+    function updateFee(uint256 _feePercentage) external onlyOwner {
+        feePercentage = _feePercentage;
+
+        emit FeeUpdated(feePercentage);
+    }
+
+    /// @notice Add a strategy to the allowlist
+    /// @param _strategy The address of the strategy
+    function addToApprovedStrategies(address _strategy) external onlyOwner {
+        approvedStrategies[_strategy] = true;
+        usedStrategies[_strategy] = true;
+    }
+
+    /// @notice Remove a strategy from the allowlist
+    /// @param _strategy The address of the strategy
+    function removeFromApprovedStrategies(address _strategy) external onlyOwner {
+        approvedStrategies[_strategy] = false;
+    }
+
+    /// @notice Add a strategy to the used list
+    /// @param _strategy The address of the strategy
+    function addToUsedStrategies(address _strategy) external onlyOwner {
+        usedStrategies[_strategy] = true;
+    }
+
+    /// ====================================
+    /// ======= Internal Functions =========
+    /// ====================================
+
+    /// @notice Deduct the fee and transfers the amount to the distribution strategy
+    /// @param _token The address of the token to transfer
+    /// @param _amount The amount to transfer
+    /// @param _poolId The pool id
+    /// @param _distributionStrategy The address of the distribution strategy
+    function _fundPool(address _token, uint256 _amount, uint256 _poolId, address _distributionStrategy) internal {
+        uint256 feeAmount = (_amount * feePercentage) / FEE_DENOMINATOR;
+
+        // Pay the protocol fee
+        _transferAmount(treasury, feeAmount, _token);
+
+        // Send the remaining amount to the distribution strategy
+        uint256 amountAfterFee = _amount - feeAmount;
+        _transferAmount(payable(_distributionStrategy), amountAfterFee, _token);
+
+        emit PoolFunded(_poolId, amountAfterFee, feeAmount);
+    }
+
+    /// @notice Transfers the amount to the address
+    /// @param _to The address to transfer to
+    /// @param _amount The amount to transfer
+    /// @param _token The address of the token to transfer
+    function _transferAmount(address payable _to, uint256 _amount, address _token) internal {
+        if (_token == address(0)) {
+            // Native Token
+            (bool sent,) = _to.call{value: _amount}("");
+            if (!sent) {
+                revert TRANSFER_FAILED();
+            }
+        } else {
+            // ERC20 Token
+            IERC20Upgradeable(_token).transfer(_to, _amount);
+        }
```

Comment:
I recommend using `safeTransferLib` here. They handle all the weird ERC20 edge cases where things break. Check out this list to see how wonky it can get: https://github.com/d-xo/weird-erc20

Reply:
fixed via 0eb884141e1bb0eeb745aff14cd96342330aa891